src/main/kotlin/org/rust/lang/core/macros/MacroExpansionManager.kt

```diff
@@ -620,9 +620,10 @@ private class MacroExpansionServiceImplInner(
 
     fun getExpandedFrom(element: RsExpandedElement): RsMacroCall? {
         checkReadAccessAllowed()
-        if (element.stubParent is RsFile /*TODO*/) {
-            val file = element.containingFile.virtualFile ?: return null
-            if (isExpansionFile(file)) {
+        val parent = element.stubParent
+        if (parent is RsFile) {
+            val file = parent.virtualFile ?: return null
```

Comment:
```suggestion
            val file = parent.virtualFile as? VirtualFileWithId ?: return null
```

Reply:
This doesn't work because we don't want to cast the variable to `VirtualFileWithId`, we want just check that it is `VirtualFileWithId` and use it as `VirtualFile`

Reply:
Indeed!